docs/TCP_Network.txt

```diff
@@ -1,100 +1,99 @@
-DHT. 
+The public/private key pair the TCP server uses is the same one he uses for the
+DHT.
 
-all crypto for communication with the server uses the crypto_box() function of 
+All crypto for communication with the server uses the crypto_box() function of
 NaCl.
 
-TCP doesn't have packets so what we will refer to as packets are sent this way: 
+TCP doesn't have packets so what we will refer to as packets are sent this way:
 [[uint16_t (length of data)][data]]
 
 So if you would inspect the TCP stream you would see:
-[[uint16_t (length of data)][data]][[uint16_t (length of 
+[[uint16_t (length of data)][data]][[uint16_t (length of
 data)][data]][[uint16_t (length of data)][data]]
 
-Note that both handshake packets don't have this format (the length for them is 
+Note that both handshake packets don't have this format (the length for them is
 always the same so we don't need to specify it.)
 
 When the client connects to the server, he sends this packet:
-[public key of client (32 bytes)][nonce for the encrypted data [24 
-bytes]][encrypted with the private key of the client and public key of the 
-server and the nonce:[public key (32 bytes) and][base nonce we want the server 
-to use to encrypt the packets sent to us (24 bytes)]]
+[temporary public key of client (32 bytes)][nonce for the encrypted data [24
+bytes]][encrypted with the temporary private key of the client and public key of
+the server and the nonce:[public key of the client (32 bytes) and][base nonce
+the client will use to to encrypt the packets sent to the server (24 bytes)]]
 
 The server responds with:
-[nonce for the encrypted data [24 bytes]][encrypted with the public key of the 
-client and private key of the server and the nonce:[public key (32 bytes) 
-and][base nonce we want the client to use to encrypt the packets sent to us (24 
-bytes)]]
+[nonce for the encrypted data [24 bytes]][encrypted with the temporary public
+key of the client and private key of the server and the nonce:[temporary public
+key of the server (32 bytes) and][base nonce the server will use to encrypt the
+packets sent to the client (24 bytes)]]
 
-All packets to the server are end to end encrypted with the information 
-received 
-(and sent) in the handshake.
+All packets to the server are end to end encrypted with the information
+received (and sent) in the handshake.
 
-(first packet is encrypted with the base nonce the private key for which the 
-client sent the server the public key and the public key we sent to the client, 
+(first packet is encrypted with the base nonce the private key for which the
+client sent the server the public key and the public key we sent to the client,
 the next with base nonce + 1...)
 
-The connection is set to an unconfirmed state until a packet is received and 
+The connection is set to an unconfirmed state until a packet is received and
 decrypted correctly using the information in the handshake.
```

Comment:
Actually, the connection is set to an confirmed state when the handshake was successfull and not after the first packet was received and decrypted.